myapp/views/IndexView.py

```diff
@@ -29,9 +32,22 @@ def get(self, request: HttpRequest) -> HttpResponse:
         """
 
         num_claims = Claim.objects.all().count()
+        upload_form = RecordUploadForm()
+        
         context = {
-            'num_claims': num_claims
+            'num_claims': num_claims,
+            'upload_form': upload_form,
         }
 
         logger.info(f"{request.user} accessed the index page.")
         return render(request, self.template_name, context=context)
+
+    @login_required
+    @require_http_methods(["POST"])
+    def record_upload(request):
+        form = RecordUploadForm(request.POST, request.FILES)
+        if form.is_valid():
+            file = request.FILES['file']
+            UploadedRecord.upload_claims_from_file(file, None)
+                
+        return HttpResponseRedirect("/")
```

Comment:
This could be modified to better align with class-based views if that's what we want to (mostly) stick to. They support HTTP methods, and where these aren't possible we could use function-based
```suggestion
    @method_decorator(login_required, name="dispatch")
    def post(self, request: HttpRequest) -> HttpResponse:
        form = RecordUploadForm(request.POST, request.FILES)
        if form.is_valid():
            file = request.FILES['file']
            UploadedRecord.upload_claims_from_file(file, None)

        return redirect("/")
```

Reply:
Good shout, I've moved the upload functionality out of the index page into CustomerDashBoardView.py and it should be more in line with the class based views